torch/nn/parallel/distributed.py

```diff
@@ -225,8 +250,9 @@ def _sync_params(self):
                 for tensor, param in zip(tensors, module.parameters()):
                     param.data.set_(tensor)
 
+        # module buffer sync
         buffers = list(self.module._all_buffers())
-        if len(buffers) > 0:
+        if self.broadcast_buffers and len(buffers) > 0:
```

Comment:
You can skip making the list if this is false. It's quite expensive by itself

Reply:
changed